script.js

```diff
@@ -1,4 +1,259 @@
+
+//global variables
+var deck = [];
+var gameMode = "first round";
+var winningMode = ""; //blackjack, bigger number max 21
+var playerHand = []; //to reset
+var computerHand = [];
+var gameRound = 1;
+
+function getRandomNumber(max) {
+  return Math.floor(Math.random() * max);
+}
+function shuffleDeck() {
+  for (i = 0; i < deck.length; i += 1) {
+    var randomIndex = getRandomNumber(deck.length);
+    var currentIndex = i;
+    var currentCard = deck[currentIndex];
+    var randomCard = deck[randomIndex];
+    deck[currentIndex] = randomCard;
+    deck[randomIndex] = currentCard;
+  }
+  return deck;
+}
+
+function getStartingHand(numberOfCards, array) {
+  for (i = 0; i < numberOfCards; i += 1) {
+    var card = dealCard();
+    array.push(card);
+  }
+  return array;
+}
+
+function dealCard() {
+  var drawnCard = deck.pop();
+  return drawnCard;
+}
+
+function displayUserCards(array) {
+  var output = `drew: <br>`;
+  for (i = 0; i < array.length; i += 1) {
+    output = output + `${array[i].name} of ${array[i].suit}<br>`;
+  }
+  return output;
+}
+
+function displayAllPlayerCards() {
+  var output = `Player ${displayUserCards(
+    playerHand
+  )} <br>Computer ${displayUserCards(computerHand)}`;
+  return output;
+}
+
+// find total to determine ace rank and to determine whether dealer draws a card
+function findCardTotal(array) {
+  var cardTotal = 0;
+  for (i = 0; i < array.length; i += 1) {
+    cardTotal += array[i].rank;
+  }
+  return cardTotal;
+}
+
+function newAceTotal(array) {
+  var cardTotal = findCardTotal(array);
+  for (i = 0; i < array.length; i += 1) {
+    if (array[i].rank === 1) {
+      array[i].rank = 11;
+      var index = i;
+      cardTotal = findCardTotal(array);
+      if (cardTotal > 21) {
+        array[index].rank = 1;
+        cardTotal = findCardTotal(array);
+      }
+    }
+  }
+  return cardTotal;
+}
+
+function isMatch(number, cardTotal) {
+  return cardTotal === number;
+}
+
+function isLessThan(number, array) {
+  var cardTotal = newAceTotal(array);
+  return cardTotal < number;
+}
+
+function isBust(number, cardTotal) {
+  //Boolean
+  return cardTotal > number;
+}
+
+function checkWinningMode() {
+  var playerCardTotal = newAceTotal(playerHand);
+  var computerCardTotal = newAceTotal(computerHand);
+  var playerBlackjack = isMatch(21, playerCardTotal);
+  var computerBlackjack = isMatch(21, computerCardTotal);
+  if (isBust(21, playerCardTotal) || isBust(21, computerCardTotal)) {
+    winningMode = "bust";
+  } else if (playerBlackjack || computerBlackjack) {
+    winningMode = "blackjack";
+  } else {
+    if (gameRound === 1) {
+      gameMode = "game continues";
+      gameRound += 1;
+    } else {
+      winningMode = "bigger number";
+    }
+  }
+}
+
+function findWinner() {
+  var playerCardTotal = newAceTotal(playerHand);
+  var computerCardTotal = newAceTotal(computerHand);
+  if (playerCardTotal === computerCardTotal) {
+    var output = "Player and computer";
```

Comment:
If you want to use the same variable inside all your conditional statements, you should declare the variable outside of all the statements first. In this case, you should be declaring it above line 121. Although you code still works, that is because `var` has unusual behaviour. If you were to use `let` or `const`, you would notice it would not work due to scoping.